commons-packet/commons-packet-service/src/main/java/io/mosip/commons/packetmanager/service/PacketWriterService.java

```diff
@@ -0,0 +1,156 @@
+package io.mosip.commons.packetmanager.service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.mosip.commons.packet.constants.PacketUtilityErrorCodes;
+import io.mosip.commons.packet.dto.TagDeleteResponseDto;
+import io.mosip.commons.packet.dto.TagDto;
+import io.mosip.commons.packet.dto.TagRequestDto;
+import io.mosip.commons.packet.dto.TagResponseDto;
+import io.mosip.commons.packet.exception.TagCreationException;
+import io.mosip.commons.packet.exception.TagDeletionException;
+import io.mosip.commons.packet.facade.PacketReader;
+import io.mosip.commons.packet.facade.PacketWriter;
+import io.mosip.commons.packet.util.PacketManagerLogger;
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.kernel.core.exception.BaseUncheckedException;
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+
+@Component
+public class PacketWriterService {
+    private static Logger LOGGER = PacketManagerLogger.getLogger(PacketWriterService.class);
+    
+    @Autowired
+    private PacketReader packetReader;
+    
+    @Autowired
+    private PacketWriter packetWriter;
+    
+    
+    public TagResponseDto addTags(TagDto tagDto) {
+    	try {
+			Map<String, String> newTags = new HashMap<String, String>();
+			Map<String, String> existingTags = packetReader.getTags(tagDto.getId());
+			if (existingTags.isEmpty()) {
+				newTags.putAll(tagDto.getTags());
+			} else {
+				for (Entry<String, String> entry : tagDto.getTags().entrySet()) {
+					if (existingTags.containsKey(entry.getKey())) {
+
+						throw new TagCreationException(PacketUtilityErrorCodes.TAG_ALREADY_EXIST.getErrorCode(),
+								PacketUtilityErrorCodes.TAG_ALREADY_EXIST.getErrorMessage());
+
+					} else {
+						newTags.put(entry.getKey(), entry.getValue());
+					}
+				}
+			}
+			tagDto.setTags(newTags);

```

Comment:
This code and else condition above to add is not making sense, we can use the same tags that is in tagDto

Reply:
Fix: removed newTags and used same tags that is in tagDto